### PR TITLE
Fix being unable to add a flux core to new the combat heart cybernetic

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Fix being unable to add a flux core to new the combat heart cybernetic
## Why It's Good For The Game
bugfix
## Changelog
:cl:
fix: Fix being unable to add a flux core to new the combat heart cybernetic
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
While we're touching this it should be changed to `item_interaction` or whatever the new relevant proc is

**Review by Jacquerel — APPROVED** [review]

File: code/modules/surgery/organs/internal/heart/heart_anomalock.dm [hunk 1]
```diff
@@ -114,19 +114,22 @@
...
+	. = NONE
+
+	if(!istype(tool, required_anomaly))
+		return ITEM_INTERACT_BLOCKING
 	if(core)
 		balloon_alert(user, "core already in!")
 		return
-	if(!user.transferItemToLoc(item, src))
+	if(!user.transferItemToLoc(tool, src))
 		return
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/heart/heart_anomalock.dm:126]:
```suggestion
		return ITEM_INTERACT_BLOCKING
```

File: code/modules/surgery/organs/internal/heart/heart_anomalock.dm [hunk 2]
```diff
@@ -114,19 +114,22 @@
...
-	if(!istype(item.type, required_anomaly))
-		return ..()
+/obj/item/organ/internal/heart/cybernetic/anomalock/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+
+	if(!istype(tool, required_anomaly))
+		return ITEM_INTERACT_BLOCKING
 	if(core)
 		balloon_alert(user, "core already in!")
 		return
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/heart/heart_anomalock.dm:124]:
```suggestion
		return ITEM_INTERACT_BLOCKING
```

File: code/modules/surgery/organs/internal/heart/heart_anomalock.dm [hunk 3]
```diff
@@ -114,19 +114,22 @@
...
 	return owner
 
-/obj/item/organ/internal/heart/cybernetic/anomalock/attackby(obj/item/item, mob/living/user, params)
-	if(!istype(item.type, required_anomaly))
-		return ..()
+/obj/item/organ/internal/heart/cybernetic/anomalock/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+
+	if(!istype(tool, required_anomaly))
+		return ITEM_INTERACT_BLOCKING
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/heart/heart_anomalock.dm:121]:
```suggestion
		return NONE
```

File: code/modules/surgery/organs/internal/heart/heart_anomalock.dm [hunk 4]
```diff
@@ -114,19 +114,22 @@
 /obj/item/organ/internal/heart/cybernetic/anomalock/proc/get_held_mob()
 	return owner
 
-/obj/item/organ/internal/heart/cybernetic/anomalock/attackby(obj/item/item, mob/living/user, params)
-	if(!istype(item.type, required_anomaly))
-		return ..()
+/obj/item/organ/internal/heart/cybernetic/anomalock/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	. = NONE
+
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/heart/heart_anomalock.dm:119]:
```suggestion
```

**Pull request merged**